src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -993,66 +1010,6 @@ P256Keypair::~P256Keypair()
 
 CHIP_ERROR P256Keypair::NewCertificateSigningRequest(uint8_t * out_csr, size_t & csr_length) const
 {
-#if defined(MBEDTLS_USE_TINYCRYPT)
-    CHIP_ERROR error = CHIP_NO_ERROR;
-
-    int result = 0;
-
-    size_t out_length;
-
-    const mbedtls_uecc_keypair * keypair = to_const_keypair(&mKeypair);
-
-    mbedtls_x509write_csr csr;
-    mbedtls_x509write_csr_init(&csr);
```

Comment:
I don't think NewCertificateSigningRequest will work unless you have at least some changes from default, since it does a signature using mbedTLS APIs directly.

In the comment where I asked for this to not be fully duplicated, I was alluding to the fact that it should not be complete cut'n'paste, just #ifdef the minimum needed to actually get the code to work while still reusing the main internals.

If you tested fully with the latest commit that you can go through commissioning properly, please disregard this comment. Otherwise either adapt the code, or use `GenerateCertificateSigningRequest` that doesn't use mbedTLS (which may be a slight flash hit, but likely not large)

Reply:
It does work, with and without TinyCrypt. I have tested it.
After your last comment I analyzed the functions and it works without changes. This is because the changes are done in the mbedtls fork in [pk.h](https://github.com/NXPmicro/mbedtls/blob/tinycrypt-mbedtls-2.28/include/mbedtls/pk.h#L201) and in the current file [here](https://github.com/NXPmicro/matter/blob/integrate-tinycrypt/src/crypto/CHIPCryptoPALmbedTLS.cpp#L530).